[CINES][PASTIS] Archival profile edition

naji-cines opened the pull request:
### Description

Ajout de fonctionnalité d'édition de profil d'archivage et de profil d"unité archivistique.
 - affichage des PA et PUA de l'instance de VITAM
 - création / modification de la notice associé au profil d'archivage (JSON)
 - création / modification du profil d'archivage (RNG)
 - création / modification de profil d'unité archivistique (JSON)
 - édition des règles des PA et PUA

### Deux projets principaux
#### api-pastis
C'est le back-end de Pastis  composé de 4 projets :
 - pastis : back spring (Pastis controller, Sécurité)
 - pastis-commons (DTO, manipulation du RNG, manipulation du JSON)
 - pastis-client : interface de l'api pastis
 - pastis-standalone : projet utilisé pour construire la version standalone de pastis

#### Le front
Composé de  : 
 - ui-pastis le gateway
 - ui-frontend/project/pastis l'application Angular

### Des ajouts dans le projet api-referential
Essentiellement pour manipuler (CRUD) les notices, PA et PUA de VITAM.

### Des ajouts dans la procédure de déploiement (playbook). Pas complètement testés.
 - modification de playbook
 - ajout de templates
 - scripts Mongo pour ajouter Pastis
 - ajout des certificats de pastis dans Mongo

### Tests
Un certain nombre de tests unitaires réalisés mais sans couverture exhaustive.

bbenaissa commented on api/api-pastis/pastis-client/pom.xml:
```diff
@@ -0,0 +1,105 @@
+        <dependency>
+            <groupId>fr.gouv.vitamui</groupId>
+            <artifactId>pastis-commons</artifactId>
+            <version>5.0-SNAPSHOT</version>
```
Mettre la version dans le pom parent si c'est pas fait et l'enlever ici

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/pom.xml:
```diff
@@ -0,0 +1,150 @@
+            <scope>compile</scope>
+        </dependency>
+        <!--
+                <dependency>
```
enlever les parties commentées

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/pom.xml:
```diff
@@ -0,0 +1,150 @@
+        <!--Start of Jaxb API and it's implementation and dependencies- Java11-->
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
```
Dans le pom parent, on a déja :   <version>${jaxb.version}</version> ,

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/pom.xml:
```diff
@@ -0,0 +1,150 @@
+        <dependency>
+            <groupId>com.fasterxml</groupId>
+            <artifactId>jackson-xml-databind</artifactId>
+            <version>0.6.2</version>
```
à mettre la version dans le pom parent

alexandregranier replied:
FAIT - A VERIFIER -> mis en variable dans parent

bbenaissa commented on api/api-pastis/pastis-commons/pom.xml:
```diff
@@ -0,0 +1,150 @@
+        <dependency>
+            <groupId>com.thaiopensource</groupId>
+            <artifactId>trang</artifactId>
+            <version>20091111</version>
```
à mettre la version dans le pom parent

alexandregranier replied:
FAIT - A VERIFIER -> mis en variable dans parent

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementProperties.java:
```diff
@@ -0,0 +1,148 @@
+    String documentation;
+
+    @JsonIgnore
+    Object sedaData;
```
Plusieurs remarques et questions: 

- Pas d'accesseurs sur les champ sedaData , parent et puaData (un oubli ?)
- sur les champs sedaData , parent, on a déja les getter/setter à traver @Data

alexandregranier replied:
RIEN A FAIRE - A VERIFIER annotation @Data suffisante

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementRNG.java:
```diff
@@ -0,0 +1,199 @@
+    public static ElementProperties elementStaticRoot = new ElementProperties();
+    private static Logger LOGGER = LoggerFactory.getLogger(ElementRNG.class);
+    private static long idCounter = 0;
+    String name;
```
Même question concernant les accesseurs

alexandregranier replied:
RIEN A FAIRE - A VERIFIER annotation @DaTa suffisante

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/CardinalityTagFactory.java:
```diff
@@ -0,0 +1,83 @@
+
+    @Override
+    public CardinalityTag createTag(ElementProperties node, Tag cardinalityTag, int level) {
+        if (node.getCardinality().equals(RNGConstants.Cardinality.zeroOrMore.getLabel())) {
```
une bonne pratique est d'avoir une seule sortie d'une méthode

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/CardinalityTagFactory.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    public RNGConstants.Cardinality checkCardinalityType(String cardinalityType) {
+        if (cardinalityType.equals("ZeroOrMoreTag")) {
```
une bonne pratique est d'avoir une seule sortie d'une méthode

alexandregranier replied:
METHODE SUPPRIMEE - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/PastisFactoryTagProvider.java:
```diff
@@ -0,0 +1,59 @@
+
+    public static AbstractTagFactory getFactory(String factoryType) {
+
+        if ("RngTag".equalsIgnoreCase(factoryType)) {
```
une bonne pratique est d'avoir une seule sortie d'une méthode

alexandregranier replied:
CLASSE SUPPRIMEE - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/RngTagFactory.java:
```diff
@@ -0,0 +1,234 @@
+        if (node.getCardinality() != null) {
+            CardinalityTagFactory cardinalityFactory = new CardinalityTagFactory();
+            CardinalityTag cardinalityTag = cardinalityFactory.createTag(node, parentNode, profondeur);
+            LOGGER.info("Parsing " + cardinalityTag.getTagName());
```
Un debug ?

alexandregranier replied:
FAIT - A VERIFIER - MIS AU NIVEAU DEBUG

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/RngTagFactory.java:
```diff
@@ -0,0 +1,234 @@
+            }
+        }
+
+        if (currentTag instanceof GrammarTag) {
```
C'est une grosse méthode avec plus de 200 lignes, à découper pour plus de lisibilité

alexandregranier replied:
FAIT - A VERIFIER -

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/jaxb/BaliseXML.java:
```diff
@@ -0,0 +1,342 @@
+                }
+            }
+
+            if (currentXmlTag instanceof GrammarXML) {
```
Pareil , à découper la méthode

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-client/pom.xml:
```diff
@@ -0,0 +1,105 @@
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+
```
Toutes les dépendances liées aux tests semblent inutiles, puisque aucun test n'existent sur ce module. Il faudrait donc supprimer toutes les dépendances a partir de la ligne 59.

pyXelians replied:
Avez-vous vérifier ce point ?

naji-cines replied:
oui, c'est bon

pyXelians commented on api/api-pastis/pastis-client/pom.xml:
```diff
@@ -0,0 +1,105 @@
+        <dependency>
+            <groupId>fr.gouv.vitamui.commons</groupId>
+            <artifactId>commons-api</artifactId>
+        </dependency>
```
Je ne suis pas sûre que cette dépendance est utilisée dans ce module, il faudrait la supprimer.

pyXelians replied:
Avez-vous vérifier ce point ?

naji-cines replied:
oui, c'est bon

pyXelians commented on api/api-pastis/pastis-client/pom.xml:
```diff
@@ -0,0 +1,105 @@
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-webflux</artifactId>
+        </dependency>
```
ça n'a pas l'air d'être utilisé dans le module

pyXelians replied:
Avez-vous vérifier ce point ?

naji-cines replied:
oui, c'est bon

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-client/src/main/java/fr/gouv/vitamui/pastis/client/PastisTransformationRestClient.java:
```diff
@@ -0,0 +1,169 @@
+        return new ParameterizedTypeReference<List<ProfileResponse>>() {
+        };
+    }
+
```
Il y a pas mal de warning affichés par IntelliJ, et des sonarlint a corriger. Si vous ne l'avez pas installé, il faudrait le faire. Inline variable, exception useless, Explicit type argument etc

pyXelians commented on api/api-pastis/pastis-client/src/main/java/fr/gouv/vitamui/pastis/client/PastisTransformationRestClient.java:
```diff
@@ -0,0 +1,169 @@
+
+    public ResponseEntity<ProfileResponse> loadProfile(Notice notice, ExternalHttpContext context)
+        throws IOException {
+        LOGGER.debug("Transform profile");
```
Exception useless

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementProperties.java:
```diff
@@ -0,0 +1,148 @@
+    ElementProperties parent;
+
+    List<ElementProperties> choices = new ArrayList<ElementProperties>();
+
```
sonarlint et warning intelliJ

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementRNG.java:
```diff
@@ -0,0 +1,199 @@
+        }
+    }
+
+    // Build the a tree of properties given :
```
Erreurs sonarlint et warning intelliJ a corriger dans le fichier.

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementRNG.java:
```diff
@@ -0,0 +1,199 @@
+    }
+
+    public static void setElementsForGroupOrChoice(ElementProperties parentNode, ElementRNG node) {
+
```
Cette méthode n'a pas l'air d'être utilisée

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementRNG.java:
```diff
@@ -0,0 +1,199 @@
+    public static ElementProperties buildElementPropertiesTree(ElementRNG node, int profondeur,
+        ElementProperties parentNode) {
+        ElementProperties local = new ElementProperties();
+        LOGGER.info("Generating JSON element {}", node.getName());
```
Complexité de la méthode trop élevé. Il faudrait factoriser

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/JsonFromRng.java:
```diff
@@ -0,0 +1,82 @@
+    String cardinality;
+
+    String valuleOrData;
+
```
Coquille dans le nom de la propriété

pyXelians replied:
la coquielle est toujours là => valu**L**eOrData

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/JsonFromRng.java:
```diff
@@ -0,0 +1,82 @@
+
+    public void setCardinality(String cardinality) {
+        if (null != RNGConstants.CardinalityMap.get(cardinality)) {
+            this.cardinality = RNGConstants.CardinalityMap.get(cardinality);
```
remplacer par une équation ternaire ?

alexandregranier replied:
FAIT

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/PuaData.java:
```diff
@@ -0,0 +1,58 @@
+    Integer MaxLenght;
+    Integer Minimum;
+    Integer Maximum;
+    Boolean ExclusiveMinimum;
```
UpperCase sur les attributs ?  il y a pas moyen d'éviter ça ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/CardinalityTagFactory.java:
```diff
@@ -0,0 +1,83 @@
+        if (cardinalityType.equals("ZeroOrMoreTag")) {
+            return RNGConstants.Cardinality.zeroOrMore;
+        }
+        if (cardinalityType.equals("OneOrMoreTag")) {
```
La méthode ne semble pas être utilsée

pyXelians replied:
Avez-vous vérifier ce point ?

naji-cines replied:
Elle est utilisé ici common/dto/factory/RngTagFactory.java L310

alexandregranier replied:
RIEN A FAIRE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/RngTagFactory.java:
```diff
@@ -0,0 +1,234 @@
+
+    static RngTag rngTree;
+
+
```
Beaucoup de warning et de sonarlint qui simplifierai  pas mal le code, condition always true, condition may be simplified, variable never used,  Merge this if statement with the enclosing one etc 

De plus je pense qu'il serait bien de factoriser cette méthode pour y voir plus clair

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/RngTagFactory.java:
```diff
@@ -0,0 +1,234 @@
+The fact that you are presently reading this means that you have had
+knowledge of the CeCILL-C license and that you accept its terms.
+*/
+
```
le package test n'existe pas, il faudrait le créer et surtout écrire des tests unitaires, avec la taille des méthodes et leurs compléxités il me parait essentiel d'avoir des tests pour assurer un minimum de maintenance

pyXelians replied:
Je vois le package mais pas les tests unitaires associés ?

alexandregranier replied:
REFACTO FAIT -> TESTS UNITAIRES NON FAITS

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/jaxb/BaliseXML.java:
```diff
@@ -0,0 +1,342 @@
+@Data
+@NoArgsConstructor
+public class BaliseXML {
+
```
Même remarque sur sonarlint et les warnings + tests unitaires + factorisation de la méthode

alexandregranier replied:
WARNING ET REFACTO FAIT. TU REPOUSSES

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/profiles/ProfileType.java:
```diff
@@ -0,0 +1,47 @@
+    PUA;
+
+    private ProfileType() {
+    }
```
private n'est pas nécessaire

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/pua/PuaMetadata.java:
```diff
@@ -0,0 +1,62 @@
+
+    public Map<String, PuaMetadataDetails> getPuaDetails() {
+        return puaDetails;
+    }
```
ça n'a pas l'air d'être utilisé

pyXelians replied:
Avez vous pu vérifier ce point ?

naji-cines replied:
resolved

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/seda/SedaNode.java:
```diff
@@ -0,0 +1,65 @@
+
+@Data
+public class SedaNode {
+    private String Name;
```
pas moyen d'éviter les Uppercase ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/JsonFromPUA.java:
```diff
@@ -0,0 +1,351 @@
+     *
+     * @param jsonPUA the JSON Object representing the PUA
+     * @return
+     */
```
Même remarque des sonarlint et des warnings a corriger et une méthode a découper/refactoriser + tests unitaires manquants

alexandregranier replied:
REFACTO FAIT - TU REPOUSSES

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/PuaDefinitions.java:
```diff
@@ -0,0 +1,75 @@
+    protected Map<String, Object> definitions;
+    @Value("${pua.definitions.file}")
+    private String defintionsFile;
+
```
coquille dans le nom de l'attribut, ça n'a pas l'air d'être utilisé non plus

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/PuaDefinitions.java:
```diff
@@ -0,0 +1,75 @@
+public class PuaDefinitions {
+
+    protected Map<String, Object> definitions;
+    @Value("${pua.definitions.file}")
```
Pourquoi ne pas initialiser la map ? ça permettrait d'éviter de se protéger contre le null dans le set

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/PuaDefinitions.java:
```diff
@@ -0,0 +1,75 @@
+import java.util.Map;
+
+@Service
+public class PuaDefinitions {
```
Pourquoi est-ce un service ? La classe n'a pas l'air d'être utilisé ?

pyXelians replied:
Pouvez-vous vérifier ce point ?

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/PuaFromJSON.java:
```diff
@@ -0,0 +1,117 @@
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
```
Warning et sonarlint

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/PuaPastisValidator.java:
```diff
@@ -0,0 +1,576 @@
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
```
warning + sonarlint + unit test + refactoring des méthodes trop longues

pyXelians replied:
Il faudrait des tests unitaires sur cette classe

alexandregranier replied:
FAIT - TU REPOUSSES

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/FileSystemResource.java:
```diff
@@ -0,0 +1,60 @@
+import org.springframework.core.io.ByteArrayResource;
+
+public final class FileSystemResource extends ByteArrayResource {
+
```
Lombok ? sonarlint ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/NoticeUtils.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    public static String serialiseString(Object o) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
```
sonarlint + méthode non utilisée

alexandregranier replied:
METHODE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/OrderedJSONObjectFactory.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.LinkedHashMap;
+
+public class OrderedJSONObjectFactory {
+    private static final VitamUILogger LOGGER = VitamUILoggerFactory.getInstance(OrderedJSONObjectFactory.class);
```
Class non utilisée ?

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisGetXmlJsonTree.java:
```diff
@@ -0,0 +1,144 @@
+
+    // Test an xml to json and a json to xml.
+    // TODO Move this code in test package
+    public String getJsonParsedTreeTest(ElementRNG elementRNGRoot) throws JAXBException, FileNotFoundException {
```
Faire le TODO

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisGetXmlJsonTree.java:
```diff
@@ -0,0 +1,144 @@
+
+
+    public String getXmlParsedTree(String jsonString) throws IOException {
+
```
Méthode pas utilisée

alexandregranier replied:
METHODE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisGetXmlJsonTree.java:
```diff
@@ -0,0 +1,144 @@
+
+
+    public void setJsonParsed(String jsonParsed) {
+        this.jsonParsed = jsonParsed;
```
Méthode pas utilisée

alexandregranier replied:
METHODE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisGetXmlJsonTree.java:
```diff
@@ -0,0 +1,144 @@
+        try {
+            jsonString = mapper.writeValueAsString(eparent);
+        } catch (JsonGenerationException e1) {
+            e1.printStackTrace();
```
catch mutualisable,  mauvaise pratique les printStackTrace

alexandregranier replied:
METHODE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisGetXmlJsonTree.java:
```diff
@@ -0,0 +1,144 @@
+import java.io.IOException;
+
+public class PastisGetXmlJsonTree {
+
```
sonarlint

pyXelians replied:
Il reste des erreurs sonarlint

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisSAX2Handler.java:
```diff
@@ -0,0 +1,146 @@
+     * Actions à réaliser lors de la fin du document XML.
+     */
+    public void endDocument() {
+    }
```
Méthode vide ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisSAX2Handler.java:
```diff
@@ -0,0 +1,146 @@
+import java.util.Stack;
+
+public class PastisSAX2Handler extends DefaultHandler {
+
```
sonarlint

pyXelians replied:
Il reste des sonarlint et des warnings

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PropertiesUtils.java:
```diff
@@ -0,0 +1,118 @@
+import java.net.URL;
+
+public final class PropertiesUtils {
+
```
Cette classe n'est référencé que dans les tests, il serait bien de la déplacer dans un package test

alexandregranier replied:
FAIT

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PropertiesUtils.java:
```diff
@@ -0,0 +1,118 @@
+        if (url == null) {
+            throw new FileNotFoundException(FILE_NOT_FOUND_IN_RESOURCES + resourcesFile);
+        }
+        File file;
```
Deux fois la même condition ?

alexandregranier replied:
CORRIGE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PropertiesUtils.java:
```diff
@@ -0,0 +1,118 @@
+        try {
+            stream = Thread.currentThread().getContextClassLoader().getResourceAsStream(resourcesFile);
+        } catch (final SecurityException e) {
+
```
Catch qui ne fait rien ?

alexandregranier replied:
CORRIGE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/RNGProfileValidator.java:
```diff
@@ -0,0 +1,97 @@
+
+public class RNGProfileValidator {
+
+    public static final String RNG_FACTORY = "com.thaiopensource.relaxng.jaxp.XMLSyntaxSchemaFactory";
```
même remarque utilisé que dans les tests

alexandregranier replied:
CLASSE DEPLACEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/RNGProfileValidator.java:
```diff
@@ -0,0 +1,97 @@
+        } catch (SAXException e) {
+            LOGGER.error("Malformed profile rng file", e);
+            error = "Malformed profile rng file : " + e.getMessage();
+            return false;
```
Assignation de la variable error pas jamais utilisé

alexandregranier replied:
MIS EN LOG - A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/RNGProfileValidator.java:
```diff
@@ -0,0 +1,97 @@
+
+                if (Objects.equals(element, elementName) || Objects.equals(prefix, elementPrefix)) {
+                    error += "Profile file xsd have not the xsd:schema tag name.";
+                    return true;
```
Assignation de la variable error pas jamais utilisé

alexandregranier replied:
MIS EN LOG - A VERIFIER

pyXelians commented on api/api-pastis/pastis-standalone/src/main/java/fr/gouv/vitamui/pastis/standalone/controller/PastisController.java:
```diff
@@ -0,0 +1,183 @@
+@RestController
+class PastisController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PastisController.class);
```
Sonarlint

pyXelians commented on api/api-pastis/pastis-standalone/src/main/java/fr/gouv/vitamui/pastis/standalone/controller/PastisController.java:
```diff
@@ -0,0 +1,183 @@
+        if (pa != null) {
+            return ResponseEntity.ok(pa);
+        } else {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
```
Renvoyé une erreur 500 qu'importe le soucis est une très mauvaise pratique. Les exceptions doivent être gérés au cas par cas, fonctionnelles ou techniques avec des codes erreurs les plus clairs possibles.

pyXelians replied:
Pouvez-vous prendre en compte ces remarques

alexandregranier replied:
Il s'agit d'erreur I/O ou de marshalling XML. Pas d'erreur métier associé. Difficile de faire mieux.

pyXelians commented on api/api-pastis/pastis-standalone/src/main/java/fr/gouv/vitamui/pastis/standalone/controller/PastisController.java:
```diff
@@ -0,0 +1,183 @@
+            @ServerVariable(name = "port", description = "Api port", defaultValue = "8096")})})
+@RestController
+class PastisController {
+
```
Les requestMappings peuvent être remplacé par getMapping, PostMapping etc...

pyXelians replied:
Pouvez-vous prendre en compte ces remarques ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/angular.json:
```diff
@@ -0,0 +1,181 @@
+  "newProjectRoot": "projects",
+  "projects": {
+    "vitamui-library": {
+      "projectType": "library",
```
vitamui-library est deprecated et doit prochainement être supprimé, ce n'est par normal que cette librairie soit référencée ici.

naji-cines replied:
Mais, elle existe sur le dans la structure des projets ui-frontend et on peux pas construire ui-frontend sans elle. Idem pour pastis standalone. Quand le projet ui-frontend ne dépendra plus d'elle on l'a supprimera

pyXelians replied:
Je n'ai pas vu d'endroit ou vous utilisé cette librairie en mode standalone ? si vous ne l'utilisé pas il n'y a pas de raison que vous ne puissiez pas la supprimer.

alexandregranier replied:
ON EST OBLIGE DE LA LAISSER SINON, CA NE COMPILE PLUS

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/package.json:
```diff
@@ -0,0 +1,173 @@
+{
+  "name": "vitamui-ui",
+  "version": "5.4.0-SNAPSHOT",
+  "license": "MIT",
```
Si ce fichier ne doit être utilisé que pour la partie standalone, la partie scripts ne nécessite pas tout ces éléments

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/startup.service.ts:
```diff
@@ -0,0 +1,221 @@
+const WARNING_DURATION = 2000;
+const CUSTOMER_TECHNICAL_REFERENT_KEY = 'technical-referent-email';
+const CUSTOMER_WEBSITE_URL_KEY = 'website-url';
+
```
Ce fichier risque d'être très dure a maintenir si "l'original" change

naji-cines replied:
Tout à fais d'accord, mais avez vous d'autres proposition ? le startupService est responsable de l'authentification et en me temps les couleurs de vitam-ui, donc on ai obligé de l'injecter pour avoir les couleur de vitam-ui mais en neutralisant l'authentification.

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/theme.service.ts:
```diff
@@ -0,0 +1,192 @@
+
+export interface Theme {
+  colors: {[colorId: string]: string};
+}
```
Même remarque sur la maintenance

alexandregranier replied:
PAS POSSIBLE DE FAIRE MIEUX

pyXelians commented on api/api-pastis/pastis/pom.xml:
```diff
@@ -0,0 +1,435 @@
+
+    <dependencies>
+        <!-- VITAMUI -->
+
```
Il faudrait vérifier si les dépendances sont utilisées ou pas

alexandregranier replied:
NON CRITIQUE - ON PEUT FAIRE CELA PLUS TARD

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/rest/PastisController.java:
```diff
@@ -0,0 +1,158 @@
+        if (archiveProfile != null) {
+            return ResponseEntity.ok(archiveProfile);
+        } else {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
```
même remarque concernant les exceptions et les RequestMapping

alexandregranier replied:
DEJA VU PLUS HAUT

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/service/PastisService.java:
```diff
@@ -0,0 +1,373 @@
+@Setter
+@Service
+public class PastisService {
+
```
sonarlint a revoir

pyXelians replied:
Test unitaires manquants

alexandregranier replied:
SONARLINT OK - TESTS REPOUSSES

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/service/PastisService.java:
```diff
@@ -0,0 +1,373 @@
+    @Value("${rng.base.directory}")
+    private String rngLocation;
+    @Autowired
+    private PuaPastisValidator puaPastisValidator;
```
Injection a faire par constructeur

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/service/PastisService.java:
```diff
@@ -0,0 +1,373 @@
+            response = new String(os.toByteArray(), "UTF-8");
+
+        } catch (IOException e1) {
+            e1.printStackTrace();
```
Très mauvaise pratique il faut gérer l'exception

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/service/PastisService.java:
```diff
@@ -0,0 +1,373 @@
+
+        } else {
+            notice.setId("12133411121213");
+        }
```
C'est normal cet Id en dure ?

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis/src/main/java/fr/gouv/vitamui/pastis/server/service/PastisService.java:
```diff
@@ -0,0 +1,373 @@
+
+    public ProfileResponse createProfile(String type) throws URISyntaxException, IOException {
+        Resource resource;
+        ProfileType profileType;
```
Utilisé une  Enum paramétrable serait plus simple et serait réutilisable

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented:
Il y a beaucoup d'éléments facilement corrigeable avec l'utilisation de sonarlint comme plugin sur intelliJ. Il y a pas mal de code mort, pas assez de tests unitaires et des méthodes a factoriser. Je vous suggère d'installer le plugin (vérifier votre formatage au passage) et nous continuerons la review une fois les remarques déjà présentes prises en comptes.

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementProperties.java:
```diff
@@ -0,0 +1,112 @@
+            this.cardinality = RNGConstants.getCardinalityMap().get(cardinality);
+        } else {
+            this.cardinality = cardinality;
+        }
```
Tu peux utiliser getOrDefault, ou bien utiliser une équation ternaire.

alexandregranier replied:
FAIT A VERIFIER

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/ElementProperties.java:
```diff
@@ -0,0 +1,112 @@
+        if (null != RNGConstants.getGroupOrChoiceMap().get(groupOrChoice)) {
+            this.groupOrChoice = RNGConstants.getGroupOrChoiceMap().get(groupOrChoice);
+        } else {
+            this.groupOrChoice = groupOrChoice;
```
Tu peux utiliser getOrDefault, ou bien utiliser une équation ternaire.

alexandregranier replied:
FAIT

alexandregranier replied:
REVENU EN ARRIERE : PAS PRATIQUE DANS CE CAS.

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/PuaData.java:
```diff
@@ -0,0 +1,59 @@
+public class PuaData implements Serializable {
+
+    Boolean additionalProperties;
+    List<String> Enum;
```
Nom d'attribut non conventionnel => pas d'uppercase, le nom fait référence a un mot clef java.

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/factory/ValueTagFactory.java:
```diff
@@ -0,0 +1,43 @@
+
+package fr.gouv.vitamui.pastis.common.dto.factory;
+
+public class ValueTagFactory implements AbstractTagFactory<Tag> {
```
Cette classe n'a pas l'air d'être utilisée ?

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/exception/BusinessException.java:
```diff
@@ -0,0 +1,15 @@
+public class BusinessException extends PastisException{
+    public BusinessException(String message) {
+        super(message);
+    }
```
Cela ne semble pas être utilisé

alexandregranier replied:
EXCEPTION SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/exception/PastisException.java:
```diff
@@ -0,0 +1,17 @@
+    public PastisException(String message) {
+        super(message);
+    }
+
```
Il existe déjà une classe qui joue se rôle pour tout le projet => VitamUIException. Il faudrait la réutiliser et enrichir les exceptions gérées. De plus vos exceptions doivent être ensuite gérer au niveau du RestExceptionHandler

bbenaissa replied:
+1

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/exception/TechnicalException.java:
```diff
@@ -0,0 +1,16 @@
+
+    public TechnicalException(String message) {
+        super(message);
+    }
```
Même commentaire, bien que celle-ci soit utilisé, il faudrait qu'elle s'inscrive dans la même logique que le projet. Ou bien revoir la logique pour l'ensemble des exceptions

bbenaissa replied:
+1

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/rest/RestApi.java:
```diff
@@ -0,0 +1,59 @@
+    public static final String PASTIS_DOWNLOAD_PA = "/getarchiveprofile";
+    public static final String PASTIS_CREATE_PROFILE = "/createprofile";
+    public static final String PASTIS = "/pastis";
+
```
Les apis n'ont pas l'air de respecter les conventions Rest, il faut les changer => https://blog.nicolashachet.com/developpement-php/larchitecture-rest-expliquee-en-5-regles/

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/JsonFromPUA.java:
```diff
@@ -0,0 +1,354 @@
+
+@Service
+public class JsonFromPUA {
+
```
Il faudrait ajouter des tests unitaires sur ce service qui détient pas mal de logique

alexandregranier replied:
FAIT - A VERIFIER

pyXelians commented on api/api-pastis/pastis/src/test/java/fr/gouv/vitamui/pastis/server/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,167 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si l'assertionError est remonté par le validator le code qui suit n'est pas utile

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis/src/test/java/fr/gouv/vitamui/pastis/server/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,167 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si l'assertionError est remonté par le validator le code qui suit n'est pas utile

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis/src/test/java/fr/gouv/vitamui/pastis/server/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,167 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si l'assertionError est remonté par le validator le code qui suit n'est pas utile

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis/src/test/java/fr/gouv/vitamui/pastis/server/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,167 @@
+
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
```
Si l'assertionError est remonté par le validator le code qui suit n'est pas utile

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis/src/test/java/fr/gouv/vitamui/pastis/server/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,167 @@
+
+    PuaPastisValidator puaPastisValidator = new PuaPastisValidator();
+    JsonFromPUA jsonFromPUA = new JsonFromPUA();
+
```
Les deux services devraient être testés séparément

alexandregranier replied:
CLASSE SUPPRIMEE

GiooDev commented on deployment/scripts/mongod/1.0.0/221_pastis_application_ref.js.j2:
```diff
@@ -0,0 +1,63 @@
+    "target" : "_self"
+});
+
+db.applications.update(
```
Pourquoi faire un update du même objet juste après le insert ?

GiooDev commented on deployment/scripts/mongod/1.0.0/222_pastis_profile_ref.js.j2:
```diff
@@ -0,0 +1,32 @@
+
+print("START 222_pastis_profile_ref.js");
+
+db.profiles.insert({
```
Attention à ne pas oublier de prendre en compte les organisations déjà existantes ? (notamment dans le cas des montées de version).

GiooDev replied:
Exemple de création de profils pour l'ensemble des tenants existants (fonctionne pour une installation from scratch & upgrade): https://github.com/ProgrammeVitam/vitam-ui/blob/develop/deployment/scripts/mongod/2.0.0/47_create_new_profiles_during_upgrade.js.j2

Pattern de nommage des profils à respecter: `PROFIL_<tenant.identifier>-<APP_NAME>-<SUFFIX>`
Si pas besoin de multi-profiles pour l'application, le <SUFFIX> = DEFAULT

GiooDev replied:
Ce script ne semble plus nécessaire car le script : `deployment/scripts/mongod/2.0.0/48_create_new_profiles_pastis_during_upgrade.js.j2` semble faire l'ensemble des créations des profils pour chacun des tenants.

GiooDev commented on deployment/scripts/mongod/1.0.0/307_iam_ref.js.j2:
```diff
@@ -58,4 +58,21 @@ db.groups.updateOne( {
+}, {
+    $addToSet: {
+        "profileIds":  {
+            $each: [
```
Pas besoin du each pour 1 seule valeur.

GiooDev commented on deployment/vitamui_pastis.yml:
```diff
@@ -0,0 +1,24 @@
+---
```
Déplacer ce script dans le répertoire `ansbile-vitamui/`

GiooDev commented on deployment/scripts/mongod/1.0.0/307_iam_ref.js.j2:
```diff
@@ -58,4 +58,21 @@ db.groups.updateOne( {
     }
```
Attention, dans le cas d'une montée de version, ce script qui a déjà été livré (notamment en 5.rc) ne sera donc pas rejoué.

Il vaut mieux sortir la mise à jour de ce groups dans un nouveau script afin qu'il soit pris en compte.

GiooDev commented on deployment/scripts/mongod/1.0.0/101_iam_system_demo.js:
```diff
@@ -197,7 +197,8 @@ db.groups.insert({
         "system_dsl",
```
Attention, sur une infra déjà existante (cas d'une mise à jour) ce script ne sera pas rejoué.

Il s'agit d'un script _demo donc pas très grave pour des environnements type production.

pyXelians commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/PastisCustomCharacterEscapeHandler.java:
```diff
@@ -0,0 +1,97 @@
+import java.io.Writer;
+
+public class PastisCustomCharacterEscapeHandler implements CharacterEscapeHandler {
+
```
ça serait bien d'avoir des Tests unitaires

alexandregranier replied:
TESTS REPOUSSES

pyXelians commented on api/api-pastis/pastis-commons/src/test/java/fr/gouv/vitamui/pastis/common/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,165 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si c'est le validator qui remonte l'erreur, le code plus bas peut être supprimé

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/test/java/fr/gouv/vitamui/pastis/common/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,165 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si c'est le validator qui remonte l'erreur, le code plus bas peut être supprimé

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/test/java/fr/gouv/vitamui/pastis/common/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,165 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si c'est le validator qui remonte l'erreur, le code plus bas peut être supprimé

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/test/java/fr/gouv/vitamui/pastis/common/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,165 @@
+        JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStreamPua));
+        JSONObject profileJson = new JSONObject(tokener);
+        puaPastisValidator.validatePUA(profileJson);
+        ElementProperties profile = jsonFromPUA.getProfileFromPUA(profileJson);
```
Si c'est le validator qui remonte l'erreur, le code plus bas peut être supprimé

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-commons/src/test/java/fr/gouv/vitamui/pastis/common/service/ImportPuaTest.java:
```diff
@@ -0,0 +1,165 @@
+
+
+@RunWith(SpringRunner.class)
+@TestPropertySource(locations = "/application-test.yml")
```
Pourquoi avoir utilisé springRunner et PropertySource ? ça n'a pas l'air d'être nécessaire dans les tests plus bas

alexandregranier replied:
CLASSE SUPPRIMEE

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/package.json:
```diff
@@ -0,0 +1,124 @@
+    "ng": "ng",
+    "ng-high-memory": "node --max_old_space_size=4000 ./node_modules/@angular/cli/bin/ng",
+    "start": "ng serve --proxy-config proxy.conf.json --disable-host-check --ssl --ssl-key $npm_package_pki_path/$npm_package_pki_asset.key --ssl-cert $npm_package_pki_path/$npm_package_pki_asset.crt",
+    "start:en": "ng serve --proxy-config proxy.conf.json  --configuration=en --disable-host-check --ssl --ssl-key $npm_package_pki_path/$npm_package_pki_asset.key --ssl-cert $npm_package_pki_path/$npm_package_pki_asset.crt",
```
Je ne pense pas que le start:en soit nécessaire ?

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/package.json:
```diff
@@ -0,0 +1,124 @@
+    "build:pastis": "export NODE_OPTIONS=--max_old_space_size=4096; ng build pastis --prod --i18n-locale fr",
+    "test": "ng test --watch=false",
+    "test:pastis": "ng test pastis --watch=false",
+    "test:vitamui-library": "ng test vitamui-library --watch=false",
```
Si vous n'utilisé pas vitamui-library, vous ne devriez pas non plus avoir besoin des scripts associés ici et plus bas

pyXelians commented on api/api-pastis/pastis-standalone/src/main/resources/standalone/package.json:
```diff
@@ -0,0 +1,124 @@
+    "lint": "ng lint --format=stylish",
+    "lint:vitamui-library": "ng lint vitamui-library --format=stylish",
+    "extract-translations": "ngx-translate-extract --input ./projects/portal/src --output ./projects/portal/src/assets/i18n/*.json --sort --format namespaced-json --key-as-default-value",
+    "copy-scss:vitamui-library": "cp-cli ./projects/vitamui-library/src/sass ./dist/sass",
```
avec ngx-translate, vous ne devriez pas avoir besoin de extract-translations

pyXelians commented on ui/ui-frontend/angular.json:
```diff
@@ -872,6 +872,147 @@
+          "configurations": {
+            "en": {
+              "aot": true,
+              "i18nFile": "projects/pastis/src/locale/messages.en.xlf",
```
cela ne devrait plus être nécessaire avec ngx-translate

pyXelians commented on ui/ui-frontend/angular.json:
```diff
@@ -872,6 +872,147 @@
+        },
+        "extract-i18n": {
+          "builder": "@angular-devkit/build-angular:extract-i18n",
+          "options": {
```
cela ne devrait plus être nécessaire avec ngx-translate

pyXelians commented on ui/ui-frontend/package.json:
```diff
@@ -29,11 +30,12 @@
+    "build:archive-search": "export NODE_OPTIONS=--max_old_space_size=4096; ng build archive-search --prod --i18n-locale fr",
+    "build:referential": "export NODE_OPTIONS=--max_old_space_size=4096; ng build referential --prod --i18n-locale fr",
+    "build:pastis": "export NODE_OPTIONS=--max_old_space_size=4096; ng build pastis --prod --i18n-locale fr",
     "analyze-portal": "ng build portal --stats-json --prod ; webpack-bundle-analyzer dist/portal/stats-es2015.json",
```
le --i18n-locale fr ne devrait plus être nécessaire avec ngx-translate

pyXelians commented on ui/ui-frontend/package.json:
```diff
@@ -64,6 +68,7 @@
     "i18n:archive-search": "ng xi18n archive-search --outputPath=src/locale --i18n-locale=fr && xliffmerge --profile projects/archive-search/xliffmerge.json fr en",
     "i18n:referential": "ng xi18n referential --outputPath=src/locale --i18n-locale=fr && xliffmerge --profile projects/referential/xliffmerge.json fr en",
+    "i18n:pastis": "ng xi18n pastis --outputPath=src/locale --i18n-locale=fr && xliffmerge --profile projects/pastis/xliffmerge.json fr en",
     "check:demo": "npm run lint:demo && npm run build:demo",
```
cela ne devrait plus être nécessaire avec ngx-translate

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/api/api.service.spec.ts:
```diff
@@ -0,0 +1,60 @@
+
+  it('should be created', () => {
+    const service: PastisApiService = TestBed.get(PastisApiService);
+    expect(service).toBeTruthy();
```
ça ne test pas grand chose; il faudrait soit compléter le test ou bien le supprimer (pour éviter qu'il consomme du temps de build pour très peu de valeur ajouté)

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/file.service.spec.ts:
```diff
@@ -0,0 +1,74 @@
+  }));
+
+  it('should be created', () => {
+    const service: FileService = TestBed.get(FileService);
```
a supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/notice.service.spec.ts:
```diff
@@ -0,0 +1,72 @@
+
+  it('should be created', () => {
+    const service: NoticeService = TestBed.get(NoticeService);
+    expect(service).toBeTruthy();
```
a supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/notice.service.ts:
```diff
@@ -0,0 +1,197 @@
+      }
+    );
+  }
+
```
Les models devraient être définis en dehors du service. De plus pour simplifier le mapping vous pouvez utiliser les affectations déstructurées : https://www.digitalocean.com/community/tutorials/understanding-destructuring-rest-parameters-and-spread-syntax-in-javascript-fr

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/popup.service.spec.ts:
```diff
@@ -0,0 +1,49 @@
+
+  it('should be created', () => {
+    const service: PopupService = TestBed.get(PopupService);
+    expect(service).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.spec.ts:
```diff
@@ -0,0 +1,64 @@
+
+  it('should be created', () => {
+    const service: ProfileService = TestBed.get(ProfileService);
+    expect(service).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+
+@Injectable({
+  providedIn: 'root'
+})
```
Supprimer le code mort et les console.log

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+  subscription3$: Subscription;
+  subscription4$: Subscription;
+  subscriptions: Subscription[] = [];
+
```
Les souscriptions ne doivent pas être géré manuellement, pour éviter cette gestion vous pouvez utiliser takeUnti et alléger le code, ou même créer une classe qui gère ça : https://dev.to/this-is-angular/dry-way-to-manage-subscriptions-in-angular-components-256j

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+    const params = new HttpParams().set('embedded', 'ALL');
+    // @ts-ignore
+    allProfilesPA = this.apiService.get(this.pastisConfig.getAllArchivalProfileUrl, {params});
+    return allProfilesPA;
```
Tu peux inline et supprimer la déclaration de la variable

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+    const params = new HttpParams().set('embedded', 'ALL');
+
+    allProfilesPUA = this.apiService.get(this.pastisConfig.getArchivalProfileUnitUrl, {params});
+    return allProfilesPUA;
```
Tu peux inline et supprimer la déclaration de la variable

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+  }
+
+  getAllProfilesVitam(): void {
+    const profiles: ProfileDescription[] = [];
```
De ce que je vois les appels sont fait séquentiellement, hors ça n'a pas l'air d'être nécessaire puisqu'il n'y a pas de lien entre les deux appels. Il faudrait donc les réaliser parallèlement.

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+      if (profileListPUA) {
+        console.log(profileListPUA.length + 'profileListPUA');
+        profileListPUA.forEach((p) => {
+          p.type = 'PUA';
```
Pourquoi ne pas utiliser l'opperator map qui semble clairement approprié à la problématique ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+        return this.data;
+      })
+    );
+    return allProfilesPA;
```
Tu peux inline

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+        return this.data;
+      })
+    );
+    return allProfilesPUA;
```
Tu peux inline

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+      if (data) {
+        // console.error(data  + " data PA")
+        data.forEach(p => p.type = 'PA');
+        data.forEach(p => tabVide.push(p));
```
Pourquoi parcourir deux fois la collection ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/profile.service.ts:
```diff
@@ -0,0 +1,309 @@
+
+  getAllProfilesPaginated(pageRequest: PageRequest): Observable<ProfileDescription[]> {
+    const tabVide: ProfileDescription[] = [];
+
```
tabVide n'a pas l'air d'être utilisé, tu peux le supprimer et par la même occasion de ne pas la remplir si tu ne l'utilises pas après

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.spec.ts:
```diff
@@ -0,0 +1,49 @@
+
+  it('should be created', () => {
+    const service: SedaService = TestBed.get(SedaService);
+    expect(service).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+  providedIn: 'root'
+})
+export class SedaService {
+
```
tslint et indentation

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+          const result = this.getSedaNode(currentChild, nameNode);
+          // Return the result if the node has been found
+          if (result) {
+            return result;
```
Avoir un seul return dans la méthode permettrai d'éviter ça

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+          const result = this.getSedaNodeRecursively(currentChild, nameNode);
+          // Return the result if the node has been found
+          if (result) {
+            return result;
```
Avoir un seul return dans la méthode permettrai d'éviter ça

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+    for (const child of sedaNode.Children) {
+      const nodeFound = this.getSedaNodeCollection(child, nodeName, collectionName);
+      if (nodeFound) {
+        return nodeFound;
```
Avoir un seul return dans la méthode permettrai d'éviter ça

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+    const fileNodesNames = fileNode.children.map(e => e.name);
+    const allowedSelectableList = sedaNode.Children.filter(x => (!fileNodesNames.includes(x.Name) &&
+                                                      x.Cardinality !== CardinalityConstants.Obligatoire.valueOf())
+                                                      ||
```
la condition est trop complexe il faudrait la découper en predicate peut être ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/seda.service.ts:
```diff
@@ -0,0 +1,228 @@
+  }
+
+  isDuplicated(nomDuChamp: string, sedaParent: SedaData) {
+    if (sedaParent.Name == nomDuChamp) {
```
nomDuChamp => français ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/toggle-sidenav.service.spec.ts:
```diff
@@ -0,0 +1,49 @@
+
+  it('should be created', () => {
+    const service: ToggleSidenavService = TestBed.get(ToggleSidenavService);
+    expect(service).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/core/services/toggle-sidenav.service.ts:
```diff
@@ -0,0 +1,57 @@
+
+    show() {
+        this.isOpened.next(true);
+        console.log('SideNav is opened');
```
supprimer les console.log

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/models/metadata-validator.ts:
```diff
@@ -0,0 +1,52 @@
+  }
+
+  static date(c: FormControl) {
+    const dateRegEx = new RegExp(/^\d{1,2}\.\d{1,2}\.\d{4}$/);
```
le type date ne suffit pas ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/pastis/pastis.module.ts:
```diff
@@ -0,0 +1,114 @@
+    TableFilterModule,
+    VitamUILibraryModule,
+    MatIconModule,
+    MatTabsModule
```
Il faudrait plutôt utiliser PastisMaterialModule si j'ai bien compris la logique

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/create-profile/create-profile.component.spec.ts:
```diff
@@ -0,0 +1,78 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/edit-profile.component.spec.ts:
```diff
@@ -0,0 +1,85 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/attributes/attributes.service.spec.ts:
```diff
@@ -0,0 +1,49 @@
+  it('should be created', () => {
+    const service: EditAttributesService = TestBed.get(EditAttributesService);
+    expect(service).toBeTruthy();
+  });
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/attributes/attributes.service.ts:
```diff
@@ -0,0 +1,46 @@
+})
+export class EditAttributesService {
+
+  constructor() { }
```
A supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/center-matmenu.directive.spec.ts:
```diff
@@ -0,0 +1,10 @@
+    // @ts-ignore
+    directive = new CenterMatmenuDirective();
+    expect(directive).toBeTruthy();
+  });
```
A compléter ou supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.component.spec.ts:
```diff
@@ -0,0 +1,102 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.component.ts:
```diff
@@ -0,0 +1,693 @@
+  this.popupSousTitre = this.translated('.POPUP_SOUS_TITRE');
+  this.popupValider = this.translated('.POPUP_VALIDER');
+  this.popupAnnuler = this.translated('.POPUP_ANNULER');
+}
```
Pourquoi ne pas avoir utiliser translate instant ? pour la pop up ça devrait suffire

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.component.ts:
```diff
@@ -0,0 +1,693 @@
+  @Output()
+  public duplicateNode: EventEmitter<FileNode> = new EventEmitter<FileNode>();
+
+  private _fileServiceSubscription: Subscription;
```
Même remarque que plus haut pour gérer les subscriptions : https://dev.to/this-is-angular/dry-way-to-manage-subscriptions-in-angular-components-256j

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.component.ts:
```diff
@@ -0,0 +1,693 @@
+
+  setNodeValue(metadata: MetadataHeaders, newValue: string) {
+    console.log(metadata.cardinalite + 'new Value ' + newValue);
+    if (newValue != null) {
```
supprimer le console.log

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.service.spec.ts:
```diff
@@ -0,0 +1,49 @@
+  it('should be created', () => {
+    const service: FileTreeMetadataService = TestBed.get(FileTreeMetadataService);
+    expect(service).toBeTruthy();
+  });
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.service.ts:
```diff
@@ -0,0 +1,254 @@
+
+          data.push({
+            id: child.id,
+            nomDuChamp: child.name,
```
On aurait pu créer un objet et un mapper pour gérer ça plutôt que de le dupliquer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.service.ts:
```diff
@@ -0,0 +1,254 @@
+    this.allowedSedaCardinalities.next(allowedCardList);
+    this.selectedCardinalities.next(this.findCardinalities(clickedNode, sedaChild, data));
+    console.log('Data on fillDataTable', data, 'with selected cards :', this.selectedCardinalities.getValue());
+    return data;
```
Les console.log sont  à supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.service.ts:
```diff
@@ -0,0 +1,254 @@
+  }
+
+  findSedaAllowedCardinalityList(sedaNode: SedaData, fileNode: FileNode): string[] {
+    let allowedCardinalityListResult: string[] = [];
```
il faudrait factoriser la méthode et diminuer le nombre de return

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.spec.ts:
```diff
@@ -0,0 +1,98 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.ts:
```diff
@@ -0,0 +1,786 @@
+    this.translatedOnChange();
+    } else if (this.isStandalone) {
+      this.notificationRemoveSuccessOne = ' a été';
+      this.notificationRemoveSuccessTwo = 'avec succès';
```
Pourquoi ne pas utiliser aussi la traduction en mode standalone ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.ts:
```diff
@@ -0,0 +1,786 @@
+    this.sedaLanguageSub = this.sedaLanguageService.sedaLanguage.subscribe(
+      (value: boolean) => {
+        this.sedaLanguage = value;
+      },
```
Il y a un changement de langue spécifique pour SEDA ?

naji-cines replied:
Oui, en langage SEDA et en francais

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.ts:
```diff
@@ -0,0 +1,786 @@
+    dataToSendToPopUp.width = '800px';
+    dataToSendToPopUp.okLabel = this.popupAddOkLabel;
+    dataToSendToPopUp.cancelLabel = this.popupAddCancelLabel;
+    dataToSendToPopUp.component = UserActionAddMetadataComponent;
```
Translate.instant aurait très bien fait l'affaire ici, donc plus besoin de conserver tous ces attributs dans le component

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.ts:
```diff
@@ -0,0 +1,786 @@
+  /** Add an item (or a list of items) in the Tree */
+  insertItem(parent: FileNode, elementsToAdd: string[], node?: FileNode, insertItemDuplicate?: boolean) {
+    console.log('After data is : %o', this.fileTreeService.nestedDataSource.data);
+    const elementsToAddFromSeda: SedaData[] = [];
```
La méthode est beaucoup trop longue il faut la factoriser

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/file-tree.component.ts:
```diff
@@ -0,0 +1,786 @@
+      }
+    }
+   /* //TODO : à revoir pour duplication des attributs
+   if(insertItemDuplicate){
```
Code mort

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/pipes/file-tree-filter.pipe.spec.ts:
```diff
@@ -0,0 +1,46 @@
+describe('FiletreeFilterPipe', () => {
+  it('create an instance', () => {
+    const pipe = new FiletreeFilterPipe();
+    expect(pipe).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree/pipes/file-tree-filter.pipe.ts:
```diff
@@ -0,0 +1,49 @@
+
+  transform(): any {
+    return null;
+  }
```
pas sûre que ce soit utile ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.spec.ts:
```diff
@@ -0,0 +1,88 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A supprimer ou compléter

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.ts:
```diff
@@ -0,0 +1,362 @@
+      constantToTranslate.call(this);
+      this.translatedOnChange();
+    } else if (this.isStandalone) {
+      this.popupCreationCancelLabel = 'Annuler';
```
même remarque concernant la traduction

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.ts:
```diff
@@ -0,0 +1,362 @@
+      this._uploadProfileSub = this.profileService.uploadProfile(formData).subscribe( (response: any) => {
+        if (response) {
+          console.log('File submited! Reponse is : ', response);
+
```
il faut supprimer tous les console.log

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.ts:
```diff
@@ -0,0 +1,362 @@
+  navigate(d: BreadcrumbDataTop) {
+    if (d.external) {
+      window.location.assign(d.url);
+    } else {
```
pourquoi on utilise un window.location ici ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.ts:
```diff
@@ -0,0 +1,362 @@
+    dataToSendToPopUp.width = '800px';
+    dataToSendToPopUp.height = '800px';
+    dataToSendToPopUp.okLabel = this.popupCreationOkLabel;
+    dataToSendToPopUp.cancelLabel = this.popupCreationCancelLabel;
```
translate instant devrait suffire ici, et donc pas besoin de déclarer autant de propriété

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/list-profile/list-profile.component.ts:
```diff
@@ -0,0 +1,362 @@
+
+@Pipe({name: 'filterByType'})
+export class FilterByTypePipe implements PipeTransform {
+  transform(listOfProfiles: ProfileDescription[], typeToFilter: string): ProfileDescription[] {
```
Il faut déclarer les pipe dans des fichiers a part

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.spec.ts:
```diff
@@ -0,0 +1,62 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A compléter ou supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.ts:
```diff
@@ -0,0 +1,144 @@
+
+  prepareSubmit(inputProfile: ProfileDescription): Observable<ProfileDescription> {
+    console.log(JSON.stringify(inputProfile) + ' inputProfile');
+
```
Supprimer les console.log

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.ts:
```diff
@@ -0,0 +1,144 @@
+  }
+
+  ngOnInit(): void {
+  }
```
Pourquoi l'avoir déclaré ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.ts:
```diff
@@ -0,0 +1,144 @@
+      this.inputProfile = this._inputProfile;
+      console.log(JSON.stringify(result));
+      this.loggingService.showSuccess('La modification du profil a bien été effectué');
+      this.profileService.refreshListProfiles();
```
élément non traduit ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.ts:
```diff
@@ -0,0 +1,144 @@
+      this.submited = false;
+      this.pending = !this.pending;
+      this.loggingService.showSuccess('Echec de la modification du profil');
+    });
```
élément non traduit ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-preview.component.spec.ts:
```diff
@@ -0,0 +1,58 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A compléter ou supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile.component.spec.ts:
```diff
@@ -0,0 +1,62 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A compléter ou supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile.component.html:
```diff
@@ -0,0 +1 @@
+<p>profile works!</p>
```
pas sûre que ce component soit utile ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/profile/profile.component.ts:
```diff
@@ -0,0 +1,52 @@
+})
+export class ProfileComponent implements OnInit {
+
+  constructor() { }
```
pas sûre que ce component soit utile ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/seda-visualizer/seda-visualizer.component.html:
```diff
@@ -0,0 +1,11 @@
+    font-family: 'Lucida Sans', Arial, sans-serif;
+    font-size: 36px; line-height: 26px; margin-top:10px;margin-bottom: 5px;">Pastis Seda Visualizer</p>
+  <div style="text-align: center; margin-top:0px">
+    <svg height=80 id="seda_legend" width=420></svg>
```
Pourquoi le style est mis en dure dans le html ?

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/seda-visualizer/seda-visualizer.component.spec.ts:
```diff
@@ -0,0 +1,72 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A compléter ou supprimer

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/seda-visualizer/seda-visualizer.component.ts:
```diff
@@ -0,0 +1,281 @@
+
+  ngOnInit() {
+   this.pastisService.getLocally(this.getSedaUrl).subscribe(sedaRules => {
+      this.sedaData = sedaRules;
```
Méthode trop longue

pyXelians commented on ui/ui-frontend/projects/pastis/src/app/shared/data-generique-popup.service.spec.ts:
```diff
@@ -0,0 +1,16 @@
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
```
A compléter ou supprimer

GiooDev commented on deployment/scripts/mongod/2.0.0/48_create_new_profiles_pastis_during_upgrade.js.j2:
```diff
@@ -0,0 +1,70 @@
+                "name": "ROLE_UPDATE_PASTIS"
+             },
+            {
+                "name": "ROLE_DELETE_PASTIS"
```
Est-ce qu'il n'y a pas besoin de créer des profils sans le droit de DELETE pour certains usages ?
Tel que pour l'app ARCHIVESEARCH ?

GiooDev commented on deployment/scripts/mongod/1.0.0/321_iam_groups_pastis_ref.js.j2:
```diff
@@ -0,0 +1,20 @@
+// ========================================= GROUPS =========================================
+
+// ----------------------------------------- LEVEL "0" -----------------------------------------
+db.groups.updateOne( {
```
Pourquoi faire ça ici alors que dans le script `deployment/scripts/mongod/2.0.0/48_create_new_profiles_pastis_during_upgrade.js.j2` on fait l'association du profil à admin_group ?

Ce script me semble inutile.

GiooDev commented on deployment/scripts/mongod/1.0.0/307_iam_ref.js.j2:
```diff
@@ -59,3 +59,5 @@ db.groups.updateOne( {
 });
 
 print("END 307_iam_ref.js");
+
```
Annuler les modifs sur ce script pour éviter le suivi dans la PR.

GiooDev commented on deployment/scripts/mongod/1.0.0/221_pastis_application_ref.js.j2:
```diff
@@ -0,0 +1,37 @@
+    "target" : "_self"
+});
+
+db.applications.update({
```
Pourquoi faire un update juste après le insert précédent ?

Modifier le insert précédent pour tout faire en 1 fois.

GiooDev commented:
Est-ce qu'il ne manquerait pas le profil de pastis dans le fichier `roles/vitamui/files/customer-init.yml` pour la création de nouvelles organisations ?

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/profiles/Notice.java:
```diff
@@ -0,0 +1,153 @@
+    private ProfileFormat format;
+
+    public Notice(Resource r) throws IOException {
+        String fileName = r.getFilename();
```
risque de NPE

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/profiles/Notice.java:
```diff
@@ -0,0 +1,153 @@
+            this.creationDate = new Timestamp(updateDate).toString();
+            this.tenant = 1;
+            this.version = 1;
+            this.name = getFileBaseName(fileName);
```
Pour name et fileBaseName : 2 appels de la méthode  getFileBaseName

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/dto/profiles/Notice.java:
```diff
@@ -0,0 +1,153 @@
+            if (getFileType(fileName).equals(ProfileType.PUA)) {
+                InputStream inputStream = getClass().getClassLoader().getResourceAsStream("rng/" +
+                    fileName);
+                JSONTokener tokener = new JSONTokener(new InputStreamReader(inputStream));
```
un check de fileExists?

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/service/JsonFromPUA.java:
```diff
@@ -0,0 +1,354 @@
+        for (String k : childPua.keySet()) {
+            switch (k) {
+                case "$ref":
+                    addPuaDataToElementIfNotPresent(childProfile);
```
les valeurs $ref, enum, pattern sont utilisés plusieurs fois, pourquoi pas une enum?

bbenaissa commented on api/api-pastis/pastis-commons/src/main/java/fr/gouv/vitamui/pastis/common/util/RNGConstants.java:
```diff
@@ -0,0 +1,404 @@
+    private static final Map<String, DataType> TypesMap = new HashMap<>();
+
+    static {
+        TypesMap.put("CodeListVersions", DataType.TOKEN);
```
Pourquoi pas externaliser ça ailleurs (un fichier de configuration) ?

bbenaissa commented on api/api-referential/referential-commons/src/main/java/fr/gouv/vitamui/referential/common/service/VitamProfileService.java:
```diff
@@ -0,0 +1,244 @@
+
+
+
+    /*
```
A supprimer

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-referential/referential-commons/src/main/java/fr/gouv/vitamui/referential/common/service/VitamArchivalProfileUnitService.java:
```diff
@@ -0,0 +1,221 @@
+package fr.gouv.vitamui.referential.common.service;
```
A ajouter le texte de la licence dans l'ensemble des nouveaux fichiers

bbenaissa commented on api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ArchivalProfileUnitExternalController.java:
```diff
@@ -0,0 +1,192 @@
+    private ArchivalProfileUnitExternalService archivalProfileUnitExternalService;
+
+    @GetMapping()
+    //@Secured(ServicesData.ROLE_GET_ARCHIVE_PROFILES_UNIT)
```
Pourquoi le controle des roles a été désactivé?

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ProfileExternalController.java:
```diff
@@ -67,11 +79,142 @@
 
     @GetMapping()
-    @Secured(ServicesData.ROLE_GET_ARCHIVE_PROFILES)
+    //@Secured(ServicesData.ROLE_GET_ARCHIVE_PROFILES)
```
Pourquoi le controle des roles a été désactivé?

bbenaissa commented on api/api-referential/referential-internal-client/src/main/java/fr/gouv/vitamui/referential/internal/client/ProfileInternalWebClient.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    @Override
+    public String getPathUrl() {
```
pourquoi l'url est null?

bbenaissa commented on api/api-referential/referential-internal/src/main/java/fr/gouv/vitamui/referential/internal/server/profile/ProfileInternalService.java:
```diff
@@ -65,27 +88,232 @@
+        }
+    }
+
+/*    public Boolean check(VitamContext vitamContext, ProfileDto archivalProfileDto) {
```
A supprimer

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on api/api-referential/referential-internal/src/main/java/fr/gouv/vitamui/referential/internal/server/rest/ProfileInternalController.java:
```diff
@@ -77,4 +95,106 @@
+
+
+
+/*    @PostMapping(CommonConstants.PATH_CHECK)
```
A supprimer

alexandregranier replied:
FAIT - A VERIFIER

bbenaissa commented on ui/ui-frontend/projects/pastis/src/app/shared/pastis-generic-popup/pastis-popup-selection.service.spec.ts:
```diff
@@ -0,0 +1,16 @@
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
```
A completer ou supprimer

bbenaissa commented on ui/ui-frontend/projects/pastis/src/app/shared/pastis-generic-popup/pastis-generic-popup.component.spec.ts:
```diff
@@ -0,0 +1,31 @@
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
A completer ou supprimer

GiooDev commented on deployment/scripts/mongod/1.0.0/221_pastis_application_ref.js.j2:
```diff
@@ -0,0 +1,30 @@
+    "hasHighlight" : false,
+    "tooltip" : "Accéder, Créer, Modifier ou Supprimer les profils d'archivage et les profils d'unité archivistique",
+    "target" : "_self",
+    {% if vitamui.pastis.base_url is defined %}
```
Pourquoi faire 2x le même if ? Est-ce qu'il ne serait pas possible de mettre dans la condition précédente les valeurs adéquates ?

GiooDev commented on deployment/scripts/mongod/1.0.0/220_cas_services_pastis_ref.js.j2:
```diff
@@ -0,0 +1,28 @@
+{% endif %}
+   "name": "Pastis Application",
+   "logoutType" : "FRONT_CHANNEL",
+{% if vitamui.pastis.base_url is defined %}
```
Pourquoi faire 2x le même if ? Est-ce qu'il ne serait pas possible de mettre dans la condition précédente les valeurs adéquates ?

bbenaissa commented:
Suite à la décision de l'admin, cette PR sera mergée et des corrections vont être ajoutés après pour améliorer la qualité et éventuellement corriger des anomlies

Commit:
Ajout de fonctionnalité d'édition de profil d'archivage et de profil d"unité archivistique.

    affichage des PA et PUA de l'instance de VITAM
    création / modification de la notice associé au profil d'archivage (JSON)
    création / modification du profil d'archivage (RNG)
    création / modification de profil d'unité archivistique (JSON)
    édition des règles des PA et PUA

Develops added by :
- Alexandre Granier
- Kevins Descamps
- Naji-cines
- Baptiste Toulemonde
